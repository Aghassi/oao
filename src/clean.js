// @flow

import path from 'path';
import rimraf from 'rimraf';
import { mainStory, chalk } from 'storyboard';
import { readAllSpecs } from './utils/readSpecs';

type Options = { src: string };

const run = async ({ src: srcPatterns }: Options) => {
  const allSpecs = await readAllSpecs(srcPatterns);
  const pkgNames = Object.keys(allSpecs);
  await Promise.all(
    pkgNames.map(
      pkgName => new Promise((resolve, reject) => {
        const { pkgPath } = allSpecs[pkgName];
        const nodeModulesPath = path.join(pkgPath, 'node_modules');
        mainStory.info(`Removing ${chalk.cyan.bold(nodeModulesPath)}...`);
        rimraf(nodeModulesPath, err => {
          if (err) {
            reject(err);
            return;
          }
          resolve();
        });
      })
    )
  );
};

export default run;
