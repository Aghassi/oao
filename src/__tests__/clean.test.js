/* eslint-env jest */
/* eslint-disable global-require, import/no-dynamic-require */

import clean from '../clean';

jest.mock('rimraf');

describe('CLEAN command', () => {
  it('executes the correct rimraf calls', async () => {
    const rimraf = require('rimraf');
    await clean({ src: 'test/fixtures/packages/*' });
    const paths = rimraf.mock.calls.map(call => call[0].replace(/\\/g, '/'));
    expect(paths).toMatchSnapshot();
  });
});
