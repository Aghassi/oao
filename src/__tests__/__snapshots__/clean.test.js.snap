// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLEAN command executes the correct rimraf calls 1`] = `
Array [
  "test/fixtures/packages/oao/node_modules",
  "test/fixtures/packages/oao-b/node_modules",
  "test/fixtures/packages/oao-c/node_modules",
  "test/fixtures/packages/oao-d/node_modules",
  "test/fixtures/packages/oao-priv/node_modules",
  "node_modules",
]
`;
